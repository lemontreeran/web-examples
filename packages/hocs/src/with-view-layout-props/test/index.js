import React from 'react'
import { mount } from 'enzyme'

import withViewLayoutProps from '../src/'

const Target = () => null

describe('withViewLayoutProps', () => {
  it('should pass props through', () => {
    const EnhancedTarget = withViewLayoutProps((state) => state)(Target)
    const wrapper = mount(
      <EnhancedTarget a={1} b={2} />
    )

    expect(wrapper.find(Target)).toMatchSnapshot()
  })

  it('should map layout dimensions to props', () => {
    const mockStateToProps = jest.fn((state) => state)
    const EnhancedTarget = withViewLayoutProps(mockStateToProps)(Target)
    const wrapper = mount(
      <EnhancedTarget />
    )

    wrapper.find(Target).prop('onLayout')({
      nativeEvent: {
        layout: {
          width: 1,
          height: 2,
          x: 3,
          y: 4
        }
      }
    })
    wrapper.update()

    expect(mockStateToProps.mock.calls).toMatchSnapshot()
    expect(wrapper.find(Target)).toMatchSnapshot()
  })

  it('should use provided custom `onLayout` handler name', () => {
    const EnhancedTarget = withViewLayoutProps(() => {}, 'onMyViewLayout')(Target)
    const wrapper = mount(
      <EnhancedTarget />
    )

    expect(wrapper.find(Target)).toMatchSnapshot()
  })

  describe('display name', () => {
    const origNodeEnv = process.env.NODE_ENV

    afterAll(() => {
      process.env.NODE_ENV = origNodeEnv
    })

    it('should wrap display name in non-production env', () => {
      process.env.NODE_ENV = 'test'

      const EnhancedTarget = withViewLayoutProps(() => {})(Target)
      const wrapper = mount(
        <EnhancedTarget />
      )

      expect(wrapper).toMatchSnapshot()
    })

    it('should not wrap display name in production env', () => {
      process.env.NODE_ENV = 'production'

      const EnhancedTarget = withViewLayoutProps(() => {})(Target)
      const wrapper = mount(
        <EnhancedTarget />
      )

      expect(wrapper).toMatchSnapshot()
    })
  })
})
