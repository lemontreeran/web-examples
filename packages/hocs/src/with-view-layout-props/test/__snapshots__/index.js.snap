// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withViewLayoutProps display name should not wrap display name in production env 1`] = `
<WithViewLayoutProps>
  <Target
    onLayout={[Function]}
  />
</WithViewLayoutProps>
`;

exports[`withViewLayoutProps display name should wrap display name in non-production env 1`] = `
<withViewLayoutProps(Target)>
  <Target
    onLayout={[Function]}
  />
</withViewLayoutProps(Target)>
`;

exports[`withViewLayoutProps should map layout dimensions to props 1`] = `
Array [
  Array [
    Object {},
  ],
  Array [
    Object {
      "height": 2,
      "width": 1,
      "x": 3,
      "y": 4,
    },
  ],
]
`;

exports[`withViewLayoutProps should map layout dimensions to props 2`] = `
<Target
  height={2}
  onLayout={[Function]}
  width={1}
  x={3}
  y={4}
/>
`;

exports[`withViewLayoutProps should pass props through 1`] = `
<Target
  a={1}
  b={2}
  onLayout={[Function]}
/>
`;

exports[`withViewLayoutProps should use provided custom \`onLayout\` handler name 1`] = `
<Target
  onMyViewLayout={[Function]}
/>
`;
